### PR TITLE
drivers:adc:ad7124: add register methods wrappers

### PR DESCRIPTION
**Pull request by adrimbarean**
Add register get and set functions wrappers to give them an appropriate
modern signature and call scheme.

Using the old get/set register functions means accessing the driver
handlers inner workings which is usually not desirable. Those wrappers
avoid this issue and maintain backwards compatibility.

Add convenient functions to the driver to use with IIO driver.

Signed-off-by: Andrei Drimbarean <Andrei.Drimbarean@analog.com>

### CODE REVIEW

**Comment by adrimbarean** [comment]
V1:
 - apply astyle

**Review by amiclaus — COMMENTED** [review]
nitpick in commit description:
`connversion`->`conversion`

**Comment by adrimbarean** [comment]
V2:
 - correct commit message typo

File: drivers/adc/ad7124/ad7124.c [hunk 1]
```diff
@@ -403,6 +439,27 @@ int32_t ad7124_read_data(struct ad7124_dev *dev,
...
 
+/**
+ * @brief Get the ID of the channel of the latest conversion.
+ *
+ * @param dev     - The handler of the instance of the driver.
+ * @param status  - Pointer to store the read data.
+ *
+ * @return Returns 0 for success or negative error code.
+ */
+int32_t ad7124_get_read_chan_id(struct ad7124_dev *dev, uint32_t *status)
```
Comment (amiclaus) [inline drivers/adc/ad7124/ad7124.c:450]: Maybe move function this function is separate commit since it is out of the commit's subject?

It is not quite a wrapper, rather a function that uses wrappers and adds new functionality to the driver.

File: drivers/adc/ad7124/ad7124.c [hunk 2]
```diff
@@ -212,6 +212,27 @@ int32_t ad7124_read_register(struct ad7124_dev *dev,
...
 }
 
+/**
+ * @brief Wrap the read register function to give it a modern signature.
+ * @param [in] dev - Driver handler pointer.
+ * @param [in] reg - Address of the register to be read.
+ * @param [out] readval - Pointer to the register value.
+ * @return SUCCESS in case of success, error code otherwise.
+ */
+int32_t ad7124_read_register_wrap(struct ad7124_dev *dev, uint32_t reg,
```
Comment (mchindri) [inline drivers/adc/ad7124/ad7124.c:222]: ad7124_read_register_wrap ->  ad7124_read_register2 wholdn't sound better?
nd add to ad7124_read_register comment that this function should be used instead.

File: drivers/adc/ad7124/ad7124.c (hunk 1 above)
Comment (adrimbarean) [reply drivers/adc/ad7124/ad7124.c:450]: Makes sense. Will do.

File: drivers/adc/ad7124/ad7124.c (hunk 2 above)
Comment (adrimbarean) [reply drivers/adc/ad7124/ad7124.c:222]: Will do.

**Comment by adrimbarean** [comment]
V3:
 - rename new register get/set functions and comment the deprecated ones;
 - split the PR in 2 commits to better follow logical grouping;
 - add generic functions used in IIO driver that may be reused in the future.

**Comment by adrimbarean** [comment]
V4:
 - fix astyle.

File: drivers/adc/ad7124/ad7124.c [hunk 3]
```diff
@@ -212,10 +213,32 @@ int32_t ad7124_read_register(struct ad7124_dev *dev,
...
+{
+	int32_t ret;
+
+	ret = ad7124_read_register(dev, &dev->regs[reg]);
+	if (ret != 0)
+		return ret;
+
+	*readval = dev->regs[reg].value;
+
+	return 0;
```
Comment (amiclaus) [inline drivers/adc/ad7124/ad7124.c:234]: `return ret`?
Check throughout the entire code.

File: drivers/adc/ad7124/ad7124.c [hunk 4]
```diff
@@ -212,10 +213,32 @@ int32_t ad7124_read_register(struct ad7124_dev *dev,
...
+ * @param [out] readval - Pointer to the register value.
+ * @return SUCCESS in case of success, error code otherwise.
+ */
+int32_t ad7124_read_register2(struct ad7124_dev *dev, uint32_t reg,
+			      uint32_t *readval)
+{
+	int32_t ret;
+
+	ret = ad7124_read_register(dev, &dev->regs[reg]);
+	if (ret != 0)
```
Comment (amiclaus) [inline drivers/adc/ad7124/ad7124.c:229]: use defines instead of raw values? (`SUCCESS/FAILURE`)
Check throughout the entire code.

Comment (mchindri) [reply drivers/adc/ad7124/ad7124.c:229]: or IS_ERR_VALUE(ret)

File: drivers/adc/ad7124/ad7124.c [hunk 5]
```diff
@@ -441,6 +441,27 @@ int32_t ad7124_read_data(struct ad7124_dev *dev,
...
+ */
+int32_t ad7124_get_read_chan_id(struct ad7124_dev *dev, uint32_t *status)
+{
+	int32_t ret;
+	uint32_t reg_temp;
+
+	ret = ad7124_read_register2(dev, AD7124_STATUS_REG, &reg_temp);
+	if (ret != 0)
+		return ret;
+	*status = reg_temp &AD7124_STATUS_REG_CH_ACTIVE(0xF);
```
Comment (mchindri) [inline drivers/adc/ad7124/ad7124.c:460]:
```suggestion
	*status = reg_temp & AD7124_STATUS_REG_CH_ACTIVE(0xF);
```
Add space after &

**Review by mchindri — APPROVED** [review]

File: drivers/adc/ad7124/ad7124.c (hunk 3 above)
Comment (adrimbarean) [reply drivers/adc/ad7124/ad7124.c:234]: will do

File: drivers/adc/ad7124/ad7124.c (hunk 4 above)
Comment (adrimbarean) [reply drivers/adc/ad7124/ad7124.c:229]: Didn't want to include error.h to preserve backwards compatibility.

**Comment by adrimbarean** [comment]
V5:
 - return ret;
 - add space between operators.

File: drivers/adc/ad7124/ad7124.c (hunk 5 above)
Comment (adrimbarean) [reply drivers/adc/ad7124/ad7124.c]: done
